hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsOutputStream.java

```diff
@@ -163,14 +200,93 @@ public AbfsOutputStream(AbfsOutputStreamContext abfsOutputStreamContext)
         abfsOutputStreamContext.getSasTokenRenewPeriodForStreamsInSeconds());
     this.outputStreamId = createOutputStreamId();
     this.tracingContext = new TracingContext(abfsOutputStreamContext.getTracingContext());
+    if (prefixMode == PrefixMode.BLOB) {
```

Comment:
GetBlockList needs to be called only if content length is greater than 0 (cases where the file was just created).

Reply:
Fixed.